### ISSUE TITLE
[Bug] When deleting a variant of a feature the deleted variants still render in the traffic split.

### ISSUE THREAD

**Issue by natasha-growthbook** [growthbook/growthbook#3051]
### Summary

When creating an experiment from a feature that has multiple variants and multiple traffic splits, if the variants are deleted the traffic split still renders the deleted variants.

#### Expected Behavior

The traffic split is updated to reflect the deleted variants

#### Current Behavior

The traffic shows the original configuration that includes the deleted variants causing confusion and a discrepancy (different traffic split shown on the UI of the experiment VS different traffic split shown when editing the traffic)

#### Replication Steps:

1. Create a "string" feature 
2. Create an experiment from the feature with 4 variants and 25/25/25/25 traffic split
3. Delete 2 variants, leaving 2 remaining
4. The UI will show 2 variants but 25/25/25/25 traffic split
5. Click "edit traffic" == shows 100% traffic but 25/25 split for undeleted variants

<img width="1124" alt="image" src="https://github.com/user-attachments/assets/94a9e66d-e8a2-45e5-b2ab-e72e70f60532">
<img width="784" alt="image" src="https://github.com/user-attachments/assets/e3a9c317-a997-462a-9709-ce72d45a4918">

**Comment by gavrashenko** [comment]
I'll provide some additional context regarding this issue. When it occurs, it causes a mismatch in the length of the `variations` and `weights` within the `features` array. As a result, the default weights fallback from your code is triggered:

```js
  if (weights.length !== numVariations) {
    if (process.env.NODE_ENV !== "production") {
      console.error(
        "Experiment.weights array must be the same length as Experiment.variations"
      );
    }
    weights = equal;
  }
```
This poses a significant problem, as we now have to spend extra time monitoring whether the bug occurs. 

Do you have any estimates for fixing this issue?

**Comment by gavrashenko** [comment]
Are there any updates?

**Comment by natasha-growthbook** [comment]
Apologies for any inconvenience caused by this and thank you for your ongoing patience. Checking for any updates with the team.

**Comment by natasha-growthbook** [comment]
Related : https://github.com/growthbook/growthbook/issues/2858

**Comment by lukesonnet** [comment]
@gavrashenko , we'll work on resolving this. Two solutions in the mean time:

1. Delete and recreate the experiment (not always feasible)
2. Use something like the following script to re-set the variations using our REST API. Our `GET` endpoint for experiments cleans unused variations, so you can just `POST` the `phases` object right back to the experiment to remedy this issue.

https://gist.github.com/lukesonnet/50f8958afdbcbdbfae7bba6811362981

**Comment by bryce-fitzsimons** [comment]
This should be fixed now #3317
